File: 1.two-sum.cpp
/*
 * @lc app=leetcode id=1 lang=cpp
 *
 * [1] Two Sum
 */

// @lc code=start
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> m;
        vector<int> res;
        for (int i = 0; i < nums.size(); i++) {
            if (!m.empty() && m.count(nums[i]) > 0) {
                res.push_back(m[nums[i]]);
                res.push_back(i);

                return res;
            }
            m[target - nums[i]] = i;
        }

        return res;
    }
};
// @lc code=end

File: 1.two-sum.rs
/*
 * @lc app=leetcode id=1 lang=rust
 *
 * [1] Two Sum
 */

// @lc code=start
use std::collections::HashMap;
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut m: HashMap<i32, usize> = HashMap::new();
        for i in 0..nums.len() {
            if i > 0 && m.contains_key(&nums[i]) {
                return vec!{m[&nums[i]] as i32, i as i32};
            }
            m.insert(target - nums[i], i);
        }

        return vec!{};
    }
}
// @lc code=end

